File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 1)
            return strs[0];
        string s = strs[0];
        for(auto i=1; i<strs.size(); i++){
            while(strs[i].find(s) != 0){
                s = s.substr(0, s.size()-1);
            }
        }
        return s;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
//Add characted to result that are matched in all words

class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder sb = new StringBuilder("");
        int idx = 0;
        while(idx < strs[0].length()){
            char match = strs[0].charAt(idx);
            for(int i=1; i<strs.length; i++){
                if(idx >= strs[i].length() || strs[i].charAt(idx) != match)
                    return sb.toString();
            }
            sb.append(match);
            idx++;
        }
        return sb.toString();
    }
}
